File: leetcode/Accepted/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        rev = 0
        n = x
        while x > 0:
            rev = rev * 10 + x % 10
            x = x // 10

        return rev == n

File: leetcode/Accepted/palindrome-number.cpp
class Solution {
  public:
    bool isPalindrome(int x) {
      if(x < 0) return false;
      long long tmp = x, rev = 0;
      while(tmp > 0){
        rev = rev * 10 + tmp % 10;
        tmp /= 10;
      }
      return rev == x;
    }
};
